### PR TITLE
fix(index): `tapable` deprecation warnings (`webpack >= v4.0.0`)

### PR DESCRIPTION
**Pull request by ai**
Right now the plugin shows this warning when I try to run it with webpack 4.0.0-beta2:

```
DepreciationWarning: Tappable.plugin is deprecated. Use new API on `.hooks` instead
```

This fix will support both webpack 3 and webpack 4 APIs without any warning. This way to deal with the API is recommended by @sokra https://github.com/webpack-contrib/webpack-bundle-analyzer/issues/154#issuecomment-367444806

I need this PR to add `.mjs` support to [Size Limit](https://github.com/ai/size-limit) https://github.com/ai/size-limit/pull/42

@michael-ciniawsky nice to see you here, you do a lot of work =^_^=

### CODE REVIEW

**Comment by ai** [comment]
I tested this changes in Size Limit branch and everything works.

**Comment by ai** [comment]
The diff is a little bit tricky. I just move logic to functions and make `if` with different API to bind functions: https://github.com/webpack-contrib/uglifyjs-webpack-plugin/pull/238/files#diff-1fdf421c05c1140f6d71444ea2b27638R243

File: src/index.js [hunk 1]
```diff
@@ -95,154 +95,167 @@ class UglifyJsPlugin {
...
         });
+
+        uglify.exit();
+        callback();
       });
-    });
+    };
+
+    if (compiler.hooks) {
+      const name = 'uglifyjs-webpack-plugin';
```
Comment (michael-ciniawsky) [inline src/index.js]:
```diff
- const name = 'uglifyjs-webpack-plugin';
+ const plugin = { name: 'UglifyJSPlugin' };

// ...hook.tap(plugin, cb)
```

Comment (michael-ciniawsky) [reply src/index.js]: `'\n'`

File: src/index.js [hunk 2]
```diff
@@ -95,154 +95,167 @@ class UglifyJsPlugin {
...
+      compiler.hooks.compilation.tap(name, (compilation) => {
+        if (this.options.sourceMap) {
+          compilation.hooks.buildMobule.tap(name, buildMobuleFn);
+        }
```
Comment (michael-ciniawsky) [inline src/index.js]: `'\n'`

File: src/index.js [hunk 3]
```diff
@@ -95,154 +95,167 @@ class UglifyJsPlugin {
...
+        compilation.hooks.optimizeChunkAssets.tapAsync(name, optimizeFn.bind(this, compilation));
+      });
+    } else {
+      compiler.plugin('compilation', (compilation) => {
+        if (this.options.sourceMap) {
+          compilation.plugin('build-module', buildMobuleFn);
+        }
```
Comment (michael-ciniawsky) [inline src/index.js]: `'\n'`

**Comment by ai** [comment]
@michael-ciniawsky `\n` was added, `plugin` was used, commit message was changed to common scheme

**Comment by michael-ciniawsky** [comment]
Released in [`v1.2.1`](https://github.com/webpack-contrib/uglifyjs-webpack-plugin/releases/tag/v1.2.1) 🎉  Thx

**Comment by ai** [comment]
Thanks to be sooooo fast 😺

**Comment by michael-ciniawsky** [comment]
For sure, these annoying warnings... 😛

**Pull request merged**